ci/rpm/install_maldet.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -uex
+
+source /etc/os-release
+: "${ID_LIKE:=unknown}"
+: "${ID:=unknown}"
+version="${VERSION_ID%%.*}"
+if [[ $ID_LIKE == *rhel* ]]; then
+  if [ "$version" -eq 7 ]; then
+    distro=centos
+  else
+    distro=el
+  fi
+fi
+if [[ $ID == *leap* ]]; then
+  distro=leap
+fi
+
+
+: "{BUILD_URL:=}"
+if command -v dnf; then
+  dnf --assumeyes install clamav clamav-devel
+  if [ -n "${BUILD_URL}" ]; then
+    repo_url="${BUILD_URL}artifact/artifacts/${distro}${version}/"
+    dnf --assumeyes config-manager --add-repo="$repo_url"
+    repo="${repo_url#*://}"
+    repo="${repo//%/}"
+    repo="${repo//\//_}"
+    # bug in EL7 DNF: this needs to be enabled before it can be disabled
+    dnf config-manager --save --setopt="$repo".gpgcheck=1
+    dnf config-manager --save --setopt="$repo".gpgcheck=0
+    # but even that seems to be not enough, so just brute-force it
+    if [ -d /etc/yum.repos.d ] &&
+       ! grep gpgcheck /etc/yum.repos.d/"$repo".repo; then
+        echo "gpgcheck=0" >> /etc/yum.repos.d/"$repo".repo
+    fi
+  fi
+elif command -v apt-get; then
+  apt-get --assume-yes install clamav libclamav-dev
+  service clamav-freshclam stop || true
+  systemctl disable clamav-freshclam || true
+fi
+mkdir -p /etc/clamd.d
+printf "LogSyslog yes\n" >> /etc/clamd.d/scan.conf
+
+lmd_tarball='maldetect-current.tar.gz'
+: "${JENKINS_URL:=https://build.hpdd.intel.com/}"
+
+lmd_url="${JENKINS_URL}job/daos-stack/job/tools/job/master/"
+lmd_url+="lastSuccessfulBuild/artifact/${lmd_tarball}"
+curl "${lmd_url}" --silent --show-error --fail -o "/var/tmp/${lmd_tarball}"
```

Comment:
Couldn't we just use Artifactory for all of this and have it provide the most recent successfully downloaded in the case that upstream is unavailable rather than maintaining a project and code to effectively do that?

Reply:
Or just docker layers.  I don't understand the need for all this complexity.

Reply:
Maldet is currently only available as a tarball that is periodically updated.  We cache that tarball along with other things to insulate us from internet glitches, and avoid extra downloads through firewall.

ClamAV is requiring us to use a local mirror for their malware database, we have to modify the ClamAV config file right after it is installed to use that mirror.

This script does this instead of having it duplicated in the Dockerfiles for each distro and version.